### ISSUE TITLE
No support for $DESTDIR in `make install`

### ISSUE THREAD

**Issue by xuanruiqi** [AbsInt/CompCert#290]
I'm trying to build a CompCert package for Arch Linux. To build packages, one needs a "staged install", where one `make install`s into a `DESTDIR`, and then builds a package using the virtual hierarchy tree. However, this seems not to be supported by CompCert's Makefile. This is standard among `Makefile`s for distribution.

As far as I understand, the only changes required are to prepend `$(BINDIR)`, `$(SHAREDIR)`, etc., with `$(DESTDIR)/`. Since `$DESTDIR` is unset by default, this shouldn't have an impact on anyone else, i.e. those who are not building/distributing packages.

**Comment by xuanruiqi** [comment]
I can confirm this works, so this change should be able to be incorporated seamlessly.

**Comment by bschommer** [comment]
Given the fact that CompCert is not free software I don't think support for packaging with Linux distributions is really needed. There already exists several other ways of building CompCert.

**Comment by xuanruiqi** [comment]
CompCert is not exactly free software, but under the terms of the license we are free to redistribute it under the INRIA Non-Commercial License Agreement, and for the majority of us hobbyists and researchers, good packaging support will help greatly.

Moreover, staged installs can be useful in many other cases, e.g. installing the same software to multiple servers, in an enterprise setting, so it could be useful in these circumstances, and the change is just a few lines in the Makefile.

**Comment by xavierleroy** [comment]
DESTDIR support added by @bschommer in commit 5b7fc96.

**Comment by didickman** [comment]
Just a note that this change required a very minor change from the OpenBSD packaging side.

> this shouldn't have an impact on anyone else

Not quite true. On the OpenBSD side, we were already using the configure-time -prefix option to set the staging destination. I don't know if Arch Linux could have used it as well to achieve the same purpose as us.

> DESTDIR support added by @bschommer in commit 5b7fc96.

OpenBSD also supports DESTDIR, so we ended up having the destination prefix appearing twice after this commit. To repair, we've adapted by dropping our override of -prefix at configuration time.

No complaints from our end, just wanted to note this. It does actually make our packaging marginally simpler after the change (we could delete 1 line in our Makefile), so is appreciated and the change will be reflected in our ports system after the next CompCert release.
